### PR TITLE
Features/stylist datetime selection

### PR DESCRIPTION
**Pull request by triquanminh**

### CODE REVIEW

File: src/component/bookingStep/bookingFooter.jsx [hunk 1]
```diff
@@ -1,34 +1,18 @@
...
+      className={active > 0 ? "button" : "button button--disabled"}
+      type="button"
+    >
+      Previous
+    </button>
+    <button onClick={handleNext} className="button" type="button">
+      {active < data.length - 1 ? "Next" : "Finish"}
+    </button>
+  </div>
+);
```
Comment (tranngminhdao) [inline src/component/bookingStep/bookingFooter.jsx]: Good

**Review by tranngminhdao — CHANGES_REQUESTED** [review]
Complete API request before I can approve this pull request

**Comment by triquanminh** [comment]
Ack, waiting for the BE dev to finalize the API parameters

File: src/component/bookingStep/index.jsx [hunk 2]
```diff
@@ -60,6 +60,7 @@ const BookingStep = () => {
       case 2:
         return (
           <StepDateTime
+            selectedServices={selectedServices}
             selectedDateTime={selectedDateTime}
             setSelectedDateTime={setSelectedDateTime}
```
Comment (tranngminhdao) [inline src/component/bookingStep/index.jsx:65]: Can we combine all state in step2 data

**Review by tranngminhdao — CHANGES_REQUESTED** [review]
Nested state mistake in the StepDateTime and StepServices. The father component is passing state to children, use that state, don't create another internal state

File: src/component/bookingStep/index.jsx [hunk 3]
```diff
@@ -1,36 +1,68 @@
...
+  };
 
   const handlePrev = () => {
     if (activeStep > 0) setActiveStep(activeStep - 1);
   };
 
   const handleNext = () => {
     if (activeStep === 0) {
-      const isValid = validateStepOne();
+      const isValid = validateStepOne(information);
```
Comment (tranngminhdao) [inline src/component/bookingStep/index.jsx]: Don't need to pass information params anymore

File: src/component/bookingStep/step/stepDateTime.jsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        <label className="datetime__label" htmlFor="stylist">
+          <p className="label-text">Stylist</p>
+          <select
+            name="stylist"
+            id="stylist"
+            className="datetime__select"
+            value={selDateTime?.stylistID || ""}
+            onChange={handleStylistChange}
+          >
+            {stylists !== undefined &&
```
Comment (tranngminhdao) [inline src/component/bookingStep/step/stepDateTime.jsx]: Just need to write stylists && stylists.map(.....

**Review by tranngminhdao — CHANGES_REQUESTED** [review]
Using a react-select library for the frontend customization

image

**Pull request merged**